src/Components/CreateImageWizard/formComponents/Oscap.js

```diff
@@ -30,26 +50,22 @@ const reinitDependingSteps = (change) => {
  * The selected profile will be stored in the `oscap-profile` form state variable.
  * The Component is shown or not depending on the ShowSelector variable.
  */
-const ProfileSelector = ({ input, showSelector }) => {
+const ProfileSelector = ({ input }) => {
   const { change, getState } = useFormApi();
-  const [profile, selectProfile] = useState(
-    getState()?.values?.['oscap-profile']
+  const [profile, setProfile] = useState(getState()?.values?.['oscap-profile']);
+  const [profileName, setProfileName] = useState(
+    getState()?.values?.['openscap-profile-name']
```

Comment:
In order to fix the issue, let's not use a form state here (sorry for pointing you in the wrong direction earlier). We will work with the  `useGetOscapCustomizationsQuery` hook and a `useEffect`.

The main idea is:
1. the `profileName` gets initialized with an empty string all the time.
2. `useGetOscapCustomizationsQuery` is called with the selected `profile` and distribution (same as you did at line 185)
3. then we will use a `useEffect` to propagate the data to the `profileName`

I've tested this solution on my local copy and it works well. Clicking back and forth in the wizard keeps the value, the clear button clear the field and recreating an image works fine.

If you need any help with this I'll be back on Monday to give you a hand.